gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/SubscriptionServiceImpl.java

```diff
@@ -632,6 +636,18 @@ public SubscriptionEntity process(ProcessSubscriptionEntity processSubscription,
         }
     }
 
+    private void validateSharedApiKeyProcessingRequest(ProcessSubscriptionEntity subscription) {
+        if (StringUtils.isNotEmpty(subscription.getCustomApiKey())) {
+            throw new SubscriptionNotProcessableException("Shared API keys do not allow setting a custom API key");
+        }
+        if (subscription.getStartingAt() != null) {
```

Comment:
This part of code breaks this scenario :
- 2 api, each has an API key plan
- subscribe api1 plan
- on API publisher side, in Api2 > Subscriptions, subscribe for the same application
- choose "shared api key" => error `Shared API keys do not allow setting a starting date`

(cause ApiSubscriptionResource.createSubscriptionToApi sets the startDate, and then calls `subscriptionService.process`)

Reply:
Same remark for customApiKey ;
ApiSubscriptionResource.createSubscriptionToApi allows to define a custom api key, but subsequent service denies it.